evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -1132,6 +1134,11 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> Handler for CheatcodeStackExecutor<'a
                 }
             }
 
+            // handle mocked calls
+            if let Some(ret) = self.state().mocked_calls.get(&(code_address, input.clone())) {
```

Comment:
this looks pretty expensive, not sure how to get around it thou

Reply:
which part of it you think is expensive? the clone?  have you tried without it?